sstables/compaction.cc

```diff
@@ -759,33 +793,29 @@ class regular_compaction : public compaction {
         return _creator();
     }
 
-    virtual sstable_writer* select_sstable_writer(const dht::decorated_key& dk) override {
-        if (!_writer) {
-            _sst = _creator();
-            setup_new_sstable(_sst);
+    virtual compaction_writer select_sstable_writer(const dht::decorated_key& dk) override {
```

Comment:
Not really a select anymore.

Reply:
Same comment that Raphael made, I will change this